### PR TITLE
Labs on the DataTypes 1-4 by Anton Ardyshev

### PR DESCRIPTION
**Pull request by toniardyshev**

### CODE REVIEW

File: Exercises/2-by-value.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 'use strict';
 
-const inc = null;
+const inc = n => ++n;
 
+const a = 5;
+const b = inc(a);
+console.dir({ a, b });
```
Comment (tshemsedinov) [inline Exercises/2-by-value.js]: Remove debug output

File: Exercises/3-by-reference.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
 'use strict';
 
-const inc = null;
+const inc = num => {
+  if (typeof num === 'object') ++num.n;
+};
+const obj = { n: 5 };
+inc(obj);
+console.dir(obj);
```
Comment (tshemsedinov) [inline Exercises/3-by-reference.js]: Remove debug output

File: Exercises/3-by-reference.js [hunk 3]
```diff
@@ -1,5 +1,10 @@
 'use strict';
 
-const inc = null;
+const inc = num => {
+  if (typeof num === 'object') ++num.n;
```
Comment (tshemsedinov) [inline Exercises/3-by-reference.js]: If you want to implement strict checking, why don't check field `n` existence and typeof n?

File: Exercises/4-count-types.js [hunk 4]
```diff
@@ -1,5 +1,15 @@
 'use strict';
 
-const countTypesInArray = null;
+const countTypesInArray = array => {
+  const obj = {};
+  for (const value of array) {
+    if (typeof(value) in obj) {
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]: Use `typeof value` instead of `typeof(value)`

Comment (toniardyshev) [reply Exercises/4-count-types.js]: Thank! Fixed according to comment

File: Exercises/3-by-reference.js (hunk 2 above)
Comment (toniardyshev) [reply Exercises/3-by-reference.js]: Thank! Fixed according to comment

File: Exercises/2-by-value.js (hunk 1 above)
Comment (toniardyshev) [reply Exercises/2-by-value.js]: Thank! Fixed according to comment

File: Exercises/3-by-reference.js (hunk 3 above)
Comment (toniardyshev) [reply Exercises/3-by-reference.js]: I agree. Verification was expected:
```const inc = num => {
  if (typeof num === 'object' && typeof num.n === 'number') ++num.n;
};
```
Getting an ```Error: Solution is too long```,  I decided to omit the type check.

If verification in this case is optional, can I not give it?

**Pull request closed without merging**